src/arrayMethodJoin.js

```diff
@@ -4,8 +4,27 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let mergedStr = '';
+    let separatorItem = separator;
+
+    if (separatorItem === null) {
+      separatorItem = 'null';
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        this[i] = '';
+      }
+
+      if (i === this.length - 1) {
+        mergedStr += this[i];
```

Comment:
you repeat this code 2 times
you can add the element without any conditions to result string
and add separator only if it's not the last element

Reply:
done)